src/analyses/goto_rw.h

```diff
@@ -277,8 +278,8 @@ class rw_guarded_range_set_value_sett:public rw_range_set_value_sett
 
   const guarded_range_domaint &get_ranges(objectst::const_iterator it) const
   {
-    assert(dynamic_cast<guarded_range_domaint*>(it->second)!=0);
-    return *static_cast<guarded_range_domaint*>(it->second);
+    assert(dynamic_cast<guarded_range_domaint*>(it->second.get())!=nullptr);
```

Comment:
While you're modifying the line, can this become INVARIANT or possibly one of the others?